icu4c/source/i18n/number_decimalquantity.cpp

```diff
@@ -557,26 +557,7 @@ void DecimalQuantity::_setToDecNum(const DecNum& decnum, UErrorCode& status) {
     }
 }
 
-DecimalQuantity& DecimalQuantity::setToDecNumberWithExponent(UnicodeString fractionStr, int32_t exponent, UErrorCode& status) {
-    setBcdToZero();
-    flags = 0;
-
-    // Compute the decNumber representation
-    CharString fracCharStr = CharString();
-    fracCharStr.appendInvariantChars(fractionStr, status);
-    DecNum decnum;
-    decnum.setTo(fracCharStr.toStringPiece(), status);
-    _setToDecNum(decnum, status);
-
-    int32_t numFracDigit = getVisibleFractionCount(fractionStr);
-    setMinFraction(numFracDigit);
-
-    adjustExponent(exponent);
-
-    return *this;
-}
-
-DecimalQuantity& DecimalQuantity::fromString(UnicodeString num, UErrorCode& status) {
+DecimalQuantity& DecimalQuantity::fromExponentString(UnicodeString num, UErrorCode& status) {
```

Comment:
Nit: This should either be a constructor (aka static class function) or be named `setToExponentString`.